app/src/lib/stores/app-store.ts

```diff
@@ -3576,6 +3579,21 @@ export class AppStore extends TypedBaseStore<IAppState> {
       this.emitUpdate()
     }
   }
+  private async updateStashEntryCountMetric(repository: Repository) {
+    const lastStashEntryCheck = await this.repositoriesStore.getLastStashCheckDate(
+      repository
+    )
+    const dateNow = moment()
+    const threshold = dateNow.subtract(24, 'hours')
+    if (lastStashEntryCheck == null || threshold.isAfter(lastStashEntryCheck)) {
+      // `lastStashEntryCheck` being equal to null means we've never checked for
+      // the given repo
+      const stashSize = await getStashSize(repository)
```

Comment:
We don't have any error handling in here for unexpected exit codes from the Git operation, and given this is run when the user switches repositories it could be easy to trigger? How defensive do we want to be in here?